### ISSUE TITLE
A11y Issues YR Landing Page

### ISSUE THREAD

**Issue by kelsonic** [department-of-veterans-affairs/va.gov-team#7642]
# Description

Here are the a11y issues that need to be address from the spot check review: https://github.com/department-of-veterans-affairs/va.gov-team/issues/7344#issuecomment-609955111

## Acceptance Criteria

1. (3x) Form elements **must** have labels | 508-issue-semantic-markup
    `<input class="usa-input" name="yr-search-name" type="text" value="">`
    - Potential fix: add id="yr-search-name" to the input
    `<select name="yr-search-state">`
    - Potential fix: add id="yr-search-state" to the select
    `<input class="usa-input" name="yr-search-city" type="text" value="">`
    - Potential fix: add id="yr-search-city" to the select

    See https://design.va.gov/components/form-controls

<hr>

2. Heading levels **should** only increase by one | axe-core, screenreader
    `<h3 class="vads-u-margin-top--1p5">Search criteria</h3>`
    - Potential fix: make this an `<h2>`

<hr>

3. Focus should be managed. | 508-issue-focus-mgmt, screenreader
    As Liz mentioned, focus **should** move from the Search button to the H1 on the search results page. If a search is updated on the results screen, then focus **should** move to the results heading. 

    ![Screen Shot 2020-04-06 at 10 52 04 AM](https://user-images.githubusercontent.com/57469/78590191-73bf9780-780f-11ea-9946-3cc2cad0168f.png)

<hr>

4. Zoom to 400% at 1280px — Reveals a number of cut off text areas. | zoom, 508-issue-mobile
    These **should** not be cut off, and are likely to be addressed as the mobile design/development rolls out.
[see screenshots]
    ![Screen Shot 2020-04-06 at 9 52 30 AM](https://user-images.githubusercontent.com/57469/78590018-24796700-780f-11ea-8be1-ddad5b8e933e.png)
    ![Screen Shot 2020-04-06 at 10 50 37 AM](https://user-images.githubusercontent.com/57469/78590020-24796700-780f-11ea-8cd9-7b32f9ab612e.png)
    ![Screen Shot 2020-04-06 at 10 51 08 AM](https://user-images.githubusercontent.com/57469/78590162-630f2180-780f-11ea-95ef-4a80454cda9c.png)

<hr>

5. Focus outline **should** be evident on the pagination links. This may be a sitewide design system issue. | 508-issue-focus-mgmt

    ![Screen Shot 2020-04-06 at 10 52 15 AM (2)](https://user-images.githubusercontent.com/57469/78590291-9baefb00-780f-11ea-895a-bcc490225126.png)

<hr>

6. The search results **should** read as a list for a screen reader. | 508-issue-cognition, screenreader

    **Consider** using one of these markup patterns:
    - ol or ul
    - dl, where each school is a dt with dd paired with it
    - table 

<hr>

7. For non-sighted users, consider adding the `<dfn class="sr-only">Label of what the info is</dfn>` for each item.  | 508-issue-cognition, screenreader

<hr>

8. In the keyboard test, I got stuck on the Search button. When I tabbed, I went to the content below, and it took a long time to get to the search results, which was disorienting as a sighted, keyboard and screen reader user. | 508-issue-focus-mgmt

<hr>

9. All the "heading level 4" becomes extra noisy as a screen reader user. | 508-issue-cognition, screenreader

<hr>

### Current code
```html
<div class="search-results vads-u-margin-top--2">
  <div class="medium-screen:vads-l-col vads-l-col vads-u-margin-bottom--2 vads-u-padding-x--2 vads-u-padding-y--2 vads-u-background-color--gray-light-alt vads-u-border--3px vads-u-border-color--transparent">
    <h3 class="vads-u-margin--0">Abilene Christian University</h3>
    <p class="vads-u-margin-bottom--1 vads-u-margin-top--0">Abilene, TX</p>
    <div class="vads-l-row vads-u-margin-top--2">
        <div class="vads-l-col--6 vads-u-display--flex vads-u-flex-direction--column vads-u-justify-content--space-between">
          <div class="vads-u-col">
              <h4 class="vads-u-font-family--sans vads-u-font-size--h5 vads-u-margin--0">Maximum Yellow Ribbon funding amount<br>(per student, per year)</h4>
              <p class="vads-u-margin--0">All tuition and fees not covered by Post-9/11 GI Bill benefits</p>
          </div>
          <h4 class="vads-u-font-family--sans vads-u-font-size--h5 vads-u-margin-top--2 vads-u-margin-bottom--0">Funding available for</h4>
          <p class="vads-u-margin-top--0 vads-u-margin-bottom--0">All eligible students</p>
          <h4 class="vads-u-font-family--sans vads-u-font-size--h5 vads-u-margin-top--2 vads-u-margin-bottom--0">School website</h4>
          <p class="vads-u-margin-top--0 vads-u-margin-bottom--0"><a href="www.acu.edu" rel="noreferrer noopener">www.acu.edu</a></p>
        </div>
        <div class="vads-l-col--6 vads-u-padding-left--2">
          <h4 class="vads-u-font-family--sans vads-u-font-size--h5 vads-u-margin--0">Degree type</h4>
          <p class="vads-u-margin--0">All</p>
          <h4 class="vads-u-font-family--sans vads-u-font-size--h5 vads-u-margin-top--7 vads-u-margin-bottom--0">School or program</h4>
          <p class="vads-u-margin--0">All</p>
        </div>
    </div>
  </div>

  <div class="medium-screen:vads-l-col vads-l-col vads-u-margin-bottom--2 vads-u-padding-x--2 vads-u-padding-y--2 vads-u-background-color--gray-light-alt vads-u-border--3px vads-u-border-color--transparent">
    <h3 class="vads-u-margin--0">Abraham Baldwin Agricultural College</h3>
    <p class="vads-u-margin-bottom--1 vads-u-margin-top--0">Tifton, GA</p>
    <div class="vads-l-row vads-u-margin-top--2">

<!-- etcetera -->
</div>
```

### Potential fix
```html
<dl>
    <dt role="heading" aria-level="3">
        <dfn class="sr-only">School name</dfn> 
        Abilene Christian University
    </dt>
    <dd>
        <dfn class="sr-only">City, State</dfn> 
        Abilene, <abbr title="Texas">TX</abbr>
    </dd>
    <dd>
        <dfn>Maximum Yellow Ribbon funding amount<br>(per student, per year)</dfn>
        All tuition and fees not covered by Post-9/11 GI Bill benefits
    </dd>
    <dd>
        <dfn>Funding available for</dfn>
        All eligible students</p>
    <dd>
        <dfn>School website</dfn>
        <a href="www.acu.edu" rel="noreferrer noopener">www.acu.edu</a>
    </dd>
    <dd>
        <dfn>Degree type</dfn>
        All
    </dd>
    <dd>
        <dfn>School or program</dfn>
        All
    </dd>

    <dt role="heading" aria-level="3"><dfn class="sr-only">School name</dfn> Abraham Baldwin Agricultural College</dt>
    <dd><dfn class="sr-only">City, State</dfn> Tifton, <abbr title="Georgia">GA</abbr></p>

<!-- etcetera -->
</dl> 
```

**Comment by kelsonic** [comment]
Due to the current design, I cannot use the potential fix suggested with `<dl>` tags, etc. as each school card has styling around it whereas the markup suggested above would only work with styling similar to Find VA Forms. Instead I will use `<ul>` and `<li>` tags for now! 🙂

**Comment by jenstrickland** [comment]
@kelsonic If you'd like I could hop on a call and help with getting the `<dd>` elements to layout like the design. I recommend using the `nth-child(#)` selector to pinpoint each item.

**Comment by ncksllvn** [comment]
@jenstrickland I think we would need the design to reflect the flat-list of the `<dl>` markup.

There's a project convention to use the design system utilities and grid system to implement layouts like this, and oftentimes we can get away without writing _any_ custom CSS. Writing custom CSS for a flat list of elements to create a grid like this ->

![image](https://user-images.githubusercontent.com/1915775/78714608-2e23cd00-78ea-11ea-8230-16b02711c07a.png)

Might be pretty difficult and even more so to maintain, especially with the custom media selectors we'd have to write.

**Comment by jenstrickland** [comment]
Good to know!

What's the markup that you had? It's possible to use the `<dfn>` in other places than a `<dl>`: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dfn I use it often to inject sr-only 'labels'.
